numpy/__init__.pyi

```diff
@@ -1,15 +1,42 @@
-# very simple, just enough to start running tests
-#
 import builtins
-from typing import Any, Mapping, List, Optional, Tuple, Union
+
+from typing import (
+    Any, Dict, Iterable, List, Optional, Mapping, Sized,
+    SupportsInt, SupportsFloat, SupportsComplex, SupportsBytes, SupportsAbs,
+    Tuple, Union,
+)
+
+import sys
 
 from numpy.core._internal import _ctypes
 
 _Shape = Tuple[int, ...]
 
+# Anything that can be coerced into numpy.dtype. To avoid recursive
+# definitions, any nested fields are required to be castable to a dtype object
+# are typed as Any.
+# Reference: https://docs.scipy.org/doc/numpy/reference/arrays.dtypes.html
+_ConvertibleToDtype = Union[
```

Comment:
Apparently I can actually include a forward reference to `dtype` in this top-level type alias. If that's a standard feature for pyi files, then I suppose we should make use of it rather than defining the separate `_DtypeLike` alias below.

Reply:
Nevermind, this doesn't work.

Reply:
Just pushed a commit implementing this.